### ISSUE TITLE
support for async/await

### ISSUE THREAD

**Issue by orefalo** [apparatus/concordant#2]
callback are so 70s

**Comment by pelger** [comment]
LOL - Callbacks are at least 90's be fair!

**Comment by davidmarkclements** [comment]
should be noted that for something low level and and in a potential hot path, async/await is unsuitable for performance. Currently async/await will create a minumum of four promises for each invocation, this is as per spec. Promises are heavy as it is.

**Comment by pelger** [comment]
Fair point! That being said I'm not against providing an async/await use pattern here as an alternative to the callback interface. Looking forward to the PR :)
